packages/app-config-writer/src/prompt/smartlinks-config.ts

```diff
@@ -0,0 +1,145 @@
+import type { AxiosBasicCredentials } from 'axios';
+import type { Choice } from 'prompts';
+import { prompt } from 'prompts';
+import { yellow, cyan } from 'chalk';
+import { isAppStudio } from '@sap-ux/btp-utils';
+import type { ToolsLogger } from '@sap-ux/logger';
+import { QuestionType } from '../types/prompt';
+import { t } from '..';
+import type { Service, ServiceConfig } from '..';
+import { getServices, getSystemCredentials, sendRequest } from '../smartlinks-config';
+
+/**
+ * @description Returns prompt result to provide or use service parameters
+ * @param services possible services to be offered for prompt
+ * @returns service to be used from prompt
+ */
+const getServicePrompt = async (services?: Service[]): Promise<ServiceConfig> => {
+    const service = services?.[0];
+    const { url, client } = await prompt([
+        {
+            name: QuestionType.Url,
+            type: 'text',
+            initial: service?.url,
+            message: service ? `${t('questions.service')} (${service?.source})` : `${t('questions.service')}`,
+            validate: (value: string): boolean | string => {
+                if (!value || !value.trim()) {
+                    return `${t('error.service')}`;
+                } else {
+                    return true;
+                }
+            }
+        },
+        {
+            name: QuestionType.Client,
+            type: 'text',
+            initial: service?.client,
+            message: service?.client ? `${t('questions.client')} (${service?.source})` : `${t('questions.client')}`
+        }
+    ]);
+    return { url, client };
+};
+
+/**
+ * @description Prompts the user for credentials.
+ * @param log logger to report info to the user
+ * @returns prompted user and password serialized for a basic auth header
+ */
+export async function promptUserPass(log?: ToolsLogger): Promise<AxiosBasicCredentials | undefined> {
+    if (isAppStudio()) {
+        const { authNeeded } = await prompt([
+            {
+                type: 'confirm',
+                name: 'authNeeded',
+                message: `${cyan(t('info.authNeeded'))}`
+            }
+        ]);
+        if (!authNeeded) {
+            return undefined;
+        }
+    } else {
+        log?.info(yellow(t('info.credentialsRequired')));
+    }
```

Comment:
I don't get this part. The check if authentication is needed should be independent if we are running in AppStudio or VSCode. In general authentication is then needed, when it is not provided by other means, e.g. from destination configuration or secure store.

Reply:
Reworked authentication for local store and BAS